src/gui/executionlogwidget.cpp

```diff
@@ -66,42 +61,63 @@ ExecutionLogWidget::~ExecutionLogWidget()
     delete m_ui;
 }
 
-void ExecutionLogWidget::showMsgTypes(const Log::MsgTypes &types)
+void ExecutionLogWidget::showMsgTypes(const Log::MsgTypes types)
 {
-    m_msgList->showMsgTypes(types);
+    m_msgList->setMsgTypes(types);
 }
 
 void ExecutionLogWidget::addLogMessage(const Log::Msg &msg)
 {
-    QString colorName;
-    switch (msg.type) {
-    case Log::INFO:
-        colorName = QLatin1String("blue");
-        break;
-    case Log::WARNING:
-        colorName = QLatin1String("orange");
-        break;
-    case Log::CRITICAL:
-        colorName = QLatin1String("red");
-        break;
-    default:
-        colorName = QApplication::palette().color(QPalette::WindowText).name();
-    }
-
-    const QDateTime time = QDateTime::fromMSecsSinceEpoch(msg.timestamp);
-    const QString text = QString(QLatin1String("<font color='grey'>%1</font> - <font color='%2'>%3</font>"))
-        .arg(time.toString(Qt::SystemLocaleShortDate), colorName, msg.message);
+    QDateTime time = QDateTime::fromMSecsSinceEpoch(msg.timestamp);
+    QString text = QString("%1 - %2").arg(time.toString(Qt::SystemLocaleShortDate), msg.message);
     m_msgList->appendLine(text, msg.type);
 }
 
 void ExecutionLogWidget::addPeerMessage(const Log::Peer &peer)
 {
-    const QDateTime time = QDateTime::fromMSecsSinceEpoch(peer.timestamp);
-    const QString msg = QString(QLatin1String("<font color='grey'>%1</font> - <font color='red'>%2</font>"))
-        .arg(time.toString(Qt::SystemLocaleShortDate), peer.ip);
+    QDateTime time = QDateTime::fromMSecsSinceEpoch(peer.timestamp);
+    QString text = QString("%1 - ").arg(time.toString(Qt::SystemLocaleShortDate));
+
+    if (peer.blocked)
+        text.append(tr("%1 was blocked %2", "x.y.z.w was blocked").arg(peer.ip, peer.reason));
+    else
+        text.append(tr("%1 was banned", "x.y.z.w was banned").arg(peer.ip));
 
-    const QString text = peer.blocked
-        ? tr("%1 was blocked %2", "0.0.0.0 was blocked due to reason").arg(msg, peer.reason)
-        : tr("%1 was banned", "0.0.0.0 was banned").arg(msg);
     m_peerList->appendLine(text, Log::NORMAL);
 }
+
+QList<QPair<QString, Log::MsgType>> ExecutionLogWidget::bulkLogMessages()
+{
+    const Logger *const logger = Logger::instance();
+    const QVector<Log::Msg> &messages = logger->getMessages();
```

Comment:
`logger->getMessages()` returns a variable not a reference.

Reply:
I have seen many cases in our current code that local variables are const reference initialized. Did we change views on that practice?

Since this involves an implicitly shared container it doesn't matter much in this case.

Reply:
>Did we change views on that practice?

Don't recall anything. AFAIK we only do "const reference initialize" when the function returns a const reference, for anything else a normal variable.